### PR TITLE
Fix javadoc output and warnings, add changelog, add some links to readme

### PR DESCRIPTION
**Pull request by zyro**

### CODE REVIEW

File: sdk/src/main/java/com/heroiclabs/sdk/android/entity/MatchGamer.java [hunk 1]
```diff
@@ -25,12 +25,12 @@
...
 
-    /** Gamer ID of this match participant. */
+    /** @return Gamer ID of this match participant. */
     private final String gamerId;
 
     /**
-     * The nickname of this match participant. May be a snapshot of the nickname as it was when the match started,
-     * rather than the most recent nickname for this account.
+     * @return The nickname of this match participant. May be a snapshot of the nickname as it was when the
+     *         match started, rather than the most recent nickname for this account.
```
Comment (mofirouz) [inline sdk/src/main/java/com/heroiclabs/sdk/android/entity/MatchGamer.java]: extra space?

File: README.md [hunk 2]
```diff
@@ -2,7 +2,7 @@ Heroic Labs Android SDK
 =======================
 The Android SDK for the Heroic Labs service.
 
-SDK Guide coming soon | SDK Reference coming soon
+SDK Guide coming soon | [SDK Reference](http://heroiclabs.github.io/heroiclabs-sdk-android/)
```
Comment (mofirouz) [inline README.md]: I would set the SDK guide URL as well, cause it's gonna go live very soon.

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+CHANGELOG
+=========
+
+### Unreleased changes
```
Comment (mofirouz) [inline CHANGELOG.md]: Why have the unreleased changes section?

**Comment by mofirouz** [comment]
Minor comments, otherwise :+1:

File: sdk/src/main/java/com/heroiclabs/sdk/android/entity/MatchGamer.java (hunk 1 above)
Comment (mofirouz) [inline sdk/src/main/java/com/heroiclabs/sdk/android/entity/MatchGamer.java]: Never mind, you were trying to inline it

File: README.md (hunk 2 above)
Comment (zyro) [inline README.md]: It's not live yet.

File: CHANGELOG.md (hunk 3 above)
Comment (zyro) [inline CHANGELOG.md]: It's common in changelogs I've seen to indicate upcoming changes.

File: sdk/src/main/java/com/heroiclabs/sdk/android/entity/MatchGamer.java (hunk 1 above)
Comment (zyro) [inline sdk/src/main/java/com/heroiclabs/sdk/android/entity/MatchGamer.java]: Those comments get copied directly to the getter javadoc, so I fixed the formatting.

**Pull request closed without merging**